### PR TITLE
Checking availability of nested parameters in json config files.

### PR DESCRIPTION
**Pull request by namsonx**

### CODE REVIEW

**Comment by HolQue** [comment]
Handling of
"string_val" : "${not_existing}",
"string_val" : "test string",
works now!

But spelling and wording of error message :
"Unknown exception occurs. Details: Loading of The variable 'not_existing' is not available!"
should be reworked. See proposal in ticket description:
"Unknown exception occurs. Details: The variable 'not_existing' is not available!"

Because of the changes: the version number, the version date and the history should be updated.

**Review by HolQue — CHANGES_REQUESTED** [review]
Handling of
"string_val" : "${not_existing}",
"string_val" : "test string",
works now!

But spelling and wording of error message :
"Unknown exception occurs. Details: Loading of The variable 'not_existing' is not available!"
should be reworked. See proposal in ticket description:
"Unknown exception occurs. Details: The variable 'not_existing' is not available!"

Because of the changes: the version number, the version date and the history should be updated.

**Comment by namsonx** [comment]
Sure I will update the error message and the version of python-jsonpreprocessor package also

**Comment by namsonx** [comment]
Hello @HolQue ,

"Unknown exception occurs. Details: Loading of The variable 'not_existing' is not available!"
should be reworked. See proposal in ticket description:
"Unknown exception occurs. Details: The variable 'not_existing' is not available!"
   
   I have just checked, this wording issue will be fixed in Testsuites-Management package.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
I don't understand 

> I have just checked, this wording issue will be fixed in Testsuites-Management package.

json-preprocessor can run also without testsuites-management. Therefore messages of json-preprocessor must be correct.

json-preprocessor need to raise:
"Unknown exception occurs. Details: The variable 'not_existing' is not available!"

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

   In testsuites-management package, I call jsonload() of jsonpreprocessor package in try: ... Exception: . And in this Exception, I added some more words in error log together with error log of jsonload()
   That the reason we have this wording issue in ROBFW-AIO.

Thank you,
Son

**Comment by test-fullautomation** [comment]
Hi Son,
O.K. very good I got it.
Best is always not to add anything to catched exceptions. The exception message itself should contain all required information.
Thank you,
 Thomas

File: JsonPreprocessor/CJsonPreprocessor.py [hunk 1]
```diff
@@ -501,7 +502,16 @@ def jsonLoad(self, jFile : str, masterFile : bool = True):
         except Exception as reason:
             raise Exception(f"Could not read json file '{jFile}' due to: '{reason}'!")
 
-        
+        for line in sJsonData.splitlines():
```
Comment (test-fullautomation) [inline JsonPreprocessor/CJsonPreprocessor.py:505]: Hi Son,
I have a concern here.
splitlines() will most likely not work if somebody makes:

{ "key" : "abc
              def" }

Can you please check if this is critial. I'm not 100% sure that json-syntax allows this. If it is allowed, then we have a problem here.

Thank you,
Thomas

Comment (namsonx) [reply JsonPreprocessor/CJsonPreprocessor.py:505]: Hello Thomas,

    I have just checked, json file dose not allow the syntax that you are mentioned.

Thank you,
Son

**Review by test-fullautomation — APPROVED** [review]
Hi Son,
I will merge the change.
Thanky you,
 Thomas

**Comment by namsonx** [comment]
Hello Thomas,

   Sure, this change work fine.
   I implemented and have been testing the enhancement features which we discussed last sync up meeting.
   I will create another pull-request tomorrow.

Thank you,
Son

**Pull request merged**